File: mybatis/src/main/java/data/spring/mybatis/application/provided/product/ProductUseCase.java
package data.spring.mybatis.application.provided.product;

import data.spring.mybatis.application.service.product.command.ProductSearchCommand;
import data.spring.mybatis.application.service.product.command.ProductUpdateCommand;
import data.spring.mybatis.domain.product.Product;

import java.util.List;
import java.util.Optional;

public interface ProductUseCase {
    Optional<Product> findById(Long productId);

    List<Product> findAll(ProductSearchCommand searchCommand);

    void save(Product product);

    int saveAll(List<Product> products);

    void update(ProductUpdateCommand updateCommand);

    int updateAll(List<ProductUpdateCommand> updateBatchRequest);
}

File: mybatis/src/main/kotlin/data/spring/mybatis/application/provided/product/ProductUseCase.kt
package data.spring.mybatis.application.provided.product

import data.spring.mybatis.application.service.product.command.ProductSearchCommand
import data.spring.mybatis.domain.product.Product

interface ProductUseCase : ProductUseCaseEx {
    fun findById(productId: Long): Product?

    fun findWithCond(searchCommand: ProductSearchCommand): List<Product>

    fun save(product: Product)

    fun saveAll(products: List<Product>): Int

    fun deleteAll(): Int
}